src/App/Page.php

```diff
@@ -364,6 +364,18 @@ public function run(App $app, BaseURL $baseURL, Mode $mode, Module $module, L10n
 		 */
 		$this->initContent($module, $mode);
 
+		// Load current theme info after module has been initialized as theme could have been set in module
+		$currentTheme = $app->getCurrentTheme();
```

Comment:
Ha, that's exactly one bug I found ^^ .. I also found situations where stylesheetpaths and similar things are loaded before the module call and get overridden after the module call.

This is a general design flaw that modules are executed almost at the end of an execution path but they are able to alter model data just before printing them at the frontend..

In a later stage of Friendica rework, I'd like to use Dice to create a module instance inside the index.php instead of the App. So a module would be capable to alter data at the start of a call (right after a general constructor) and the (raw)Content methods are just allowed to print data the 'right' way.

Reply:
BTW: When you are reworking the stuff: I think that we need a possibility to separate between web requests from a regular visitor, API calls and communication between the systems.

For example: Sessions are only needed with the regular visitors. And I guess it would help in debugging, when the logging would separate between these different access types.

I know that this is not really related to this PR, but I think it is not totally unrelated, since the whole theme stuff doesn't need to do on API calls and "server to server" communication as well.

Reply:
Yep, that's exactly where the road will go :-)

At the end, a module would be possible to change data like the logging-channel right through it's initialization at the beginning. So the module `Friendica\Module\Api` would alter the channel to API and disable Session-usage at initilization/constructing whereas `Friendica\Module\Display` wouldn't do that ;-)

Reply:
Sounds okay. I hope that it will be simple enough so that I will be able to understand how that works.

Reply:
If possible I'd really like to attend a next hackaton so we can discuss this whole class/constructor/Dependency stuff face to face and not through miles of explanation texts... 

Nevertheless I give my best to make it clean and understandable :)

Reply:
> Ha, that's exactly one bug I found ^^

What do you mean? Is it a change request? I'm a little confused.

Reply:
Ah sorry, I just found it nice that we both changed the code the same way because of the same reason.
-> just a comment :)